src/main/java/no/nav/veilarbaktivitet/aktiviterTilKafka/KafkaAktivitetDAO.java

```diff
@@ -0,0 +1,65 @@
+package no.nav.veilarbaktivitet.aktiviterTilKafka;
+
+import lombok.AllArgsConstructor;
+import no.nav.veilarbaktivitet.db.Database;
+import no.nav.veilarbaktivitet.domain.AktivitetStatus;
+import no.nav.veilarbaktivitet.domain.AktivitetTypeData;
+import no.nav.veilarbaktivitet.util.EnumUtils;
+import org.springframework.stereotype.Repository;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+import static no.nav.veilarbaktivitet.mappers.Helpers.typeMap;
+
+@Repository
+@AllArgsConstructor
+public class KafkaAktivitetDAO {
+    private final Database database;
+    public List<KafkaAktivitetMeldingV2> hentOppTil1000MeldingerUtenKafka() {
+        // language=sql
+        return database.query(""+
+                        " select AKTOR_ID, AKTIVITET.AKTIVITET_ID as ID, VERSJON,  FRA_DATO, TIL_DATO, ENDRET_DATO, AKTIVITET_TYPE_KODE, LIVSLOPSTATUS_KODE, AVTALT, HISTORISK_DATO " +
+                        " from AKTIVITET " +
+                        " where GJELDENDE = 1" +
+                        " and not exists(" +
+                        " select 1 from AKTIVITET_SENDT_PAA_KAFKA kafka " +
+                        " where kafka.AKTIVITET_ID = AKTIVITET.AKTIVITET_ID " +
+                        " and kafka.AKTIVITET_VERSJON = AKTIVITET.VERSJON" +
+                        " )" +
+                        " FETCH NEXT 1000 ROWS ONLY",
+                KafkaAktivitetDAO::mapKafkaAktivitetMeldingV2
```

Comment:
Om man hadde hatt unik id i Aktivitettabellen så kunne spørringen vært forenklet til where kafka.aktivitetID=aktivitet.ID. For sent for den tabellen nå men om man alltid har en unik guid id i hver tabell så er man sikret for hva som skjer i fremtiden.

Reply:
sånn som det er implementert i dag kunne man forenklet det til  `kafka.AKTIVITET_VERSJON = AKTIVITET.VERSJON `som er et evigvarende løpenummer. 
Men føler det kommer tydeligere frem hva man ønsker med denne. Hva tenker du?